istioctl/pkg/waypoint/waypoint.go

```diff
@@ -201,6 +206,9 @@ func Cmd(ctx cli.Context) *cobra.Command {
 		fmt.Sprintf("Specify the traffic type %s for the waypoint", validTrafficTypes.String()),
 	)
 
+	waypointApplyCmd.PersistentFlags().BoolVarP(&enrollNamespace, "enroll-namespace", "", false,
+		"If set, the namespace will be annotated with the waypoint name")
+
```

Comment:
If most common way to deploy waypoint is namespace scoped, should we have enroll-namespace enabled to true by default and users can opt out as needed?

Reply:
I'm curious about a few things: 
- What are the other ways to deploy the waypoint?
- How will forcing an opt-out effect those users?

cc/ @ilrudie @howardjohn

Reply:
- Other ways to deploy waypoint: users can deploy waypoint by manually creating the gateway resource (or use istioctl waypoint generate?) and apply it to the cluster.
- We could have consistent behavior between waypoint generate and apply/create for enrolling?

Reply:
 Ok, just trying to understand the alpha UX and proposed beta UX more clearly.
 
>  make it simple for users where today it is only 1 step for alpha

What exactly is only 1 step for alpha? There's a way to enroll a waypoint in a namespace other than istioctl?

> We could have consistent behavior between waypoint generate and apply/create for enrolling?

Are you suggesting that we enroll namespace by default in the generate command as well as updating `istioctl x waypoint apply` to enroll a waypoint in a namespace by default? And then, something like `istioctl x waypoint apply --unenroll-namespace`?